engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/StartEventSubscriptionManager.java

```diff
@@ -75,52 +81,61 @@ private void closeExistingMessageStartEventSubscriptions(
                 subscription.getRecord()));
   }
 
-  private DeployedProcess findLastMessageStartProcess(final ProcessMetadata processRecord) {
+  private DeployedProcess findLastStartProcess(
+      final ProcessMetadata processRecord, final Predicate<ExecutableCatchEventElement> predicate) {
     for (int version = processRecord.getVersion() - 1; version > 0; --version) {
-      final DeployedProcess lastMsgProcess =
+      final DeployedProcess lastStartProcess =
           processState.getProcessByProcessIdAndVersion(
               processRecord.getBpmnProcessIdBuffer(), version);
-      if (lastMsgProcess != null
-          && lastMsgProcess.getProcess().getStartEvents().stream()
-              .anyMatch(ExecutableCatchEventElement::isMessage)) {
-        return lastMsgProcess;
+      if (lastStartProcess != null
+          && lastStartProcess.getProcess().getStartEvents().stream().anyMatch(predicate)) {
+        return lastStartProcess;
       }
     }
 
     return null;
   }
 
-  private void openMessageStartEventSubscriptions(
+  private void openStartEventSubscriptions(
       final ProcessMetadata processRecord, final StateWriter stateWriter) {
     final long processDefinitionKey = processRecord.getKey();
     final DeployedProcess processDefinition = processState.getProcessByKey(processDefinitionKey);
     final ExecutableProcess process = processDefinition.getProcess();
     final List<ExecutableStartEvent> startEvents = process.getStartEvents();
 
-    // if startEvents contain message events
-    for (final ExecutableCatchEventElement startEvent : startEvents) {
+    for (final ExecutableStartEvent startEvent : startEvents) {
+      // if startEvent is message event
       if (startEvent.isMessage()) {
-        final ExecutableMessage message = startEvent.getMessage();
-
-        message
-            .getMessageName()
-            .map(BufferUtil::wrapString)
-            .ifPresent(
-                messageNameBuffer -> {
-                  subscriptionRecord.reset();
-                  subscriptionRecord
-                      .setMessageName(messageNameBuffer)
-                      .setProcessDefinitionKey(processDefinitionKey)
-                      .setBpmnProcessId(process.getId())
-                      .setStartEventId(startEvent.getId());
-
-                  final var subscriptionKey = keyGenerator.nextKey();
-                  stateWriter.appendFollowUpEvent(
-                      subscriptionKey,
-                      MessageStartEventSubscriptionIntent.CREATED,
-                      subscriptionRecord);
-                });
+        openMessageStartEventSubscriptions(
+            processDefinition, processDefinitionKey, startEvent, stateWriter);
       }
     }
   }
+
+  private void openMessageStartEventSubscriptions(
```

Comment:
🔧 This method only opens a single message start event subscription. Let's remove the `s` at the end of the name, to go from plural to singular form.